pagefaultgames/pokerogue
# [Misc][Bug] Add `isBatonPassable` property to `BattlerTag`s

## What are the changes?
Baton Pass is now more accurate in which properties it passes to other Pokémon.

## Why am I doing these changes?
Baton Pass currently passes `BattlerTag`s that it shouldn't to other Pokémon.

## What did change?
An `isTransferrable: boolean` property is added to `BattlerTag`s to be used by Baton Pass and the Baton item, and existing `BattlerTag`s that should be passed are marked accordingly.

## How to test the changes?
Have the opposing Pokémon apply the "infatuated" effect (or Salt Cure, or various other effects that shouldn't be passable) on your Pokémon, then use Baton Pass and verify that the effect was not passed.
`npm run test baton_pass`

## Checklist
